core/trino-main/src/test/java/io/trino/execution/TestDropNotNullConstraintConstraintTask.java

```diff
@@ -0,0 +1,172 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.trino.Session;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.TableHandle;
+import io.trino.security.AllowAllAccessControl;
+import io.trino.spi.connector.CatalogHandle;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorCapabilities;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.type.Type;
+import io.trino.sql.tree.DropNotNullConstraint;
+import io.trino.sql.tree.Identifier;
+import io.trino.sql.tree.NodeLocation;
+import io.trino.sql.tree.QualifiedName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Set;
+
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.connector.ConnectorCapabilities.NOT_NULL_COLUMN_CONSTRAINT;
+import static io.trino.spi.connector.SaveMode.FAIL;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.QueryUtil.identifier;
+import static io.trino.sql.planner.TestingPlannerContext.plannerContextBuilder;
+import static io.trino.testing.TestingHandles.TEST_CATALOG_NAME;
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDropNotNullConstraintConstraintTask
+        extends BaseDataDefinitionTaskTest
+{
+    @Override
+    @BeforeEach
+    public void setUp()
+    {
+        super.setUp();
+        metadata = new MockMetadataWithNotNull(TEST_CATALOG_NAME);
+        plannerContext = plannerContextBuilder().withMetadata(metadata).build();
+    }
+
+    @Test
+    public void testDropNotNullConstraint()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), FAIL);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).orElseThrow();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(notNullColumn("a", BIGINT), notNullColumn("b", BIGINT));
+
+        getFutureValue(executeDropNotNullConstraint(asQualifiedName(tableName), identifier("b"), false));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(notNullColumn("a", BIGINT), nullableColumn("b", BIGINT));
+    }
```

Comment:
Add an extra "executeDropNotNullConstraint" which should fail now due to the fact that the column is already nullable.